### ISSUE TITLE
incorrect SQL data type

### ISSUE THREAD

**Issue by the-Arioch** [joyfullservice/msaccess-vcs-addin#533]
Access 2013, addin versions 3/4/23 and 4.0.34

The SQL datatype must be mere "BYTE" not "unsigned byte", or the command fails

```
CREATE TABLE [Form] (
...
  [Security] UNSIGNED BYTE ,
...
```

```
<xsd:element name="Security" minOccurs="0" od:jetType="byte" od:sqlSType="tinyint" type="xsd:unsignedByte">
 <xsd:annotation>
 <xsd:appinfo>
  <od:fieldProperty name="ColumnWidth" type="3" value="-1" /> 
  <od:fieldProperty name="ColumnOrder" type="3" value="0" /> 
  <od:fieldProperty name="ColumnHidden" type="1" value="0" /> 
  <od:fieldProperty name="Description" type="10" value="0 - Free access; 1- Private; 2 - Encrypted" /> 
  <od:fieldProperty name="DecimalPlaces" type="2" value="255" /> 
  <od:fieldProperty name="DefaultValue" type="12" value="0" /> 
  <od:fieldProperty name="Required" type="1" value="0" /> 
  <od:fieldProperty name="DisplayControl" type="3" value="109" /> 
  <od:fieldProperty name="TextAlign" type="2" value="0" /> 
  <od:fieldProperty name="AggregateType" type="4" value="-1" /> 
  <od:fieldProperty name="ResultType" type="2" value="0" /> 
  <od:fieldProperty name="CurrencyLCID" type="4" value="0" /> 
  </xsd:appinfo>
  </xsd:annotation>
  </xsd:element>
```

**Comment by the-Arioch** [comment]
or, perhaps, use `TinyInt` type to be more compatible with MS SQL Server

confusingly, regardless the xxxINT naming, it is unsigned 0..255 type, thus is identical to `UNSIGNED BYTE`

OTOH it may be that some really old MDAC/Jet versions would not recognize it, yet i can hardly imaging such pre-win2000 versiosn could still be relevant today.

**Comment by joyfullservice** [comment]
Thanks for reporting this! Using `BYTE` works fine in Access 2010. (`TINYINT` doesn't) I am not concerned with supporting anything older than Access 2010, so using `BYTE` should resolve this from what I am seeing in some brief testing.  👍

**Comment by hecon5** [comment]
Question: Is this a case where conditional compilation would be appropriate as having `TINYINT` would be corret(er) for MSSQL? 
I know both "work", but we use `TINYINT` quite a lot for some of our tables (as well as unsigned bytes, which does make life odd, but I didn't make all the choices here).

**Comment by the-Arioch** [comment]
@hecon5 rather, some general setting if the export would be shaped into ANSI SQL, TransactSQL or Access/MDAC SQL

whether such a goal would ever be considered worthy

ATM the program is only concerned with Access itself, not other SQL targets.
And since there is Access version that is "tinyint"-incompatible then the choice seems obvious.

**Comment by joyfullservice** [comment]
If I am understanding the original issue correctly, I believe we were looking at the SQL generated during the TableDef export, which is intended to create Microsoft Access compatible SQL, not SQL Server SQL. At least in Access 2010, `TINYINT` is not a supported data type for creating a table from a Microsoft Access Query.

The add-in can create SQL creation statements for SQL Server tables, but that is a part of the External Databases export.

**Comment by hecon5** [comment]
I think you did understand correctly @joyfullservice. I think for the MSSQL linked queries / tables though that having it correctly name the columns would be good. For instance, we have several tables which use `TINYINT` but the Addin exports as `UNSIGNED BYTE`. 

While this doesn't really affect us too much, those folks who use the addin to version track their SQL backends with this tool may find it could cause issues depending on the use case. There doesn't appear to be a comparable value without creating another enum, so perhaps this is not the right windmill to tilt against.

**Comment by joyfullservice** [comment]
Yes, I think I see three different levels of SQL compatibility here. 

1.) For the purpose of the add-in's primary role, the export to Access compatible SQL gives the added benefit of being able to optionally use that SQL to generate simple tables. (Although the main purpose for the SQL export of TableDefs was to provide a simplified human-readable export for easy change comparison, while the XML provides the comprehensive detail needed to fully reconstruct the table during an import.)

2.) For more accurate SQL back-end tracking, the external database section utilizes SQL server (and helper scripts) to better generate SQL scripts representing those back-end objects. This is where you would find your `TINYINT` and other non-Access data types represented with actual SQL data types.

3.) If you need even better version control for back-end tables, there are a number of other 3rd party tools that specialize in this domain. For many users, I think this add-in may be adequate for their needs, but for more advanced needs other tools may be a better fit.
